toplevel/workerLoop.ml

```diff
@@ -10,6 +10,7 @@
 
 let worker_parse_extra extra_args =
   let stm_opts, extra_args = Stmargs.parse_args ~init:Stm.AsyncOpts.default_opts extra_args in
+  let extra_args = List.filter (CString.is_prefix "--kind=") extra_args in
```

Comment:
This means if we pass `--kind=nonsense` after the correct kind it will be ignored, seems wrong.